crates/core/src/lib.rs

```diff
@@ -110,12 +119,12 @@ pub use parquet;
 #[cfg(not(any(feature = "rustls", feature = "native-tls")))]
 compile_error!("You must enable at least one of the features: `rustls` or `native-tls`.");
 
-/// Creates and loads a DeltaTable from the given path with current metadata.
-/// Infers the storage backend to use from the scheme in the given table path.
+/// Creates and loads a DeltaTable from the given URL with current metadata.
+/// Infers the storage backend to use from the scheme in the given table URL.
 ///
 /// Will fail fast if specified `table_uri` is a local path but doesn't exist.
-pub async fn open_table(table_uri: impl AsRef<str>) -> Result<DeltaTable, DeltaTableError> {
-    let table = DeltaTableBuilder::from_valid_uri(table_uri)?.load().await?;
+pub async fn open_table(table_uri: Url) -> Result<DeltaTable, DeltaTableError> {
```

Comment:
Just thinking out loud .. with object store we do have `reqwest` in our dependency tree anyways, which exposes an `IntoUrl` trait ... however then we would have a coupling with the `requwest` version as we now expose its types, so not worth it.